Sources/Datadog/FeaturesIntegration/WebView/WKUserContentController+Datadog.swift

```diff
@@ -16,10 +16,24 @@ public extension WKUserContentController {
     internal func __addDatadogMessageHandler(allowedWebViewHosts: Set<String>, hostsSanitizer: HostsSanitizing) {
         let bridgeName = DatadogMessageHandler.name
 
+        let contextProvider = (Global.rum as? RUMMonitor)?.contextProvider
```

Comment:
we can inject it but as we can't unit test this method, i'm not sure what we gain by injecting it

do you mean the user should provide `contextProvider` in public API? (if so, `contextProvider` is an internal type so this shouldn't be possible)

Reply:
No, not from the user but from another source which is less transverse. This might not be possible, but assuming that a global instance is of some type could easily break.

Reply:
In other places we solve this problem by [injecting `RUMMonitor` from its `initialize()` method](https://github.com/DataDog/dd-sdk-ios/blob/0ea2b5f6307b6005ed0cc1980754ec6d2dd3cc41/Sources/Datadog/RUMMonitor.swift#L167-L168) to all integrated components. Later, this is tested respectively in each component's tests: [`URLSessionAutoInstrumentationTests`](https://github.com/DataDog/dd-sdk-ios/blob/0ea2b5f6307b6005ed0cc1980754ec6d2dd3cc41/Tests/DatadogTests/Datadog/URLSessionAutoInstrumentation/URLSessionAutoInstrumentationTests.swift#L39-L61) and [`RUMInstrumentationTests`](https://github.com/DataDog/dd-sdk-ios/blob/0ea2b5f6307b6005ed0cc1980754ec6d2dd3cc41/Tests/DatadogTests/Datadog/RUM/Instrumentation/RUMInstrumentationTests.swift#L23-L92).

I think the key element for enabling `WKUserContentController+Datadog` testability is to create separate component for locating involved services, e.g.:
```swift
internal final class WebviewsFeature {
   /// Single, shared instance of `WebviewsFeature`.
   internal static var instance: WebviewsFeature?

   let messageHandler: WKScriptMessageHandler

   func createJSBridge(for allowedWebViewHosts: Set<String>) -> WKUserScript { /*...*/ }
}
```
Going that way the the implementation of `WKUserContentController` extension will be reduced to only:
```swift
public extension WKUserContentController {
   func addDatadogMessageHandler(allowedWebViewHosts: Set<String>) {
      guard let webviewsFeature = WebviewsFeature.instance else {
         // What does it mean to have no WebviewsFeature.instance?
         // e.g. If both Logging and RUM features are disabled, will this integration be
         // still functional? We may want to log user configuration error from here.
         return
      }
      add(webviewsFeature.messageHandler, name: webviewsFeature.bridgeName)
      addUserScript(webviewsFeature.createJSBridge(for: allowedWebViewHosts))
   }
}
```
and we will be able to test everything as part of `WebviewsFeature` configuration.

Reply:
i leave `Global.rum` usage as there may be no Event Bridge installed at the time of `RUMMonitor.initialize`, `Global.rum` seems like the best option for now.

the tests requested by @ncreated are also added, in another way.